### PR TITLE
CriticNetwork Visualizer

### PR DESCRIPTION
**Pull request by Haichao-Zhang**
A function that can be used for generating the response images wrt action changes, which is useful for visualization.

### CODE REVIEW

File: alf/utils/visualizer.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        data = data.cpu().numpy()
+
+        val_img = alf.summary.render.render_heatmap(name="val_img", data=data)
+
+
+    Args:
+        net (Callable): a callable that is called as``net((obsevation, actions))``
+        observation (Tensor): [B, ...]
+        origin (tensor): tensor representing the start of the probing region,
+            with the shape of [d]
```
Comment (hnyu) [inline alf/utils/visualizer.py]: d -> action_dim

File: alf/utils/visualizer.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        dx (tensor): a tensor representing the end position of the probing
```
Comment (hnyu) [inline alf/utils/visualizer.py]: If it's the end position, then probably shouldn't be called dx.

File: alf/utils/visualizer.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        The network response image of the shape [B, K, H, W], where K denotes
+        the dimensionality of the network output for the non-batch dimension.
+    """
+
+    batch_size = observation.shape[0]
+    # total number of anchors for probing the network
+    num_anchors = H * W
+    # expand observation from [B, ...] to [B * num_anchors, ...]
+    ext_obs = torch.repeat_interleave(observation, num_anchors, dim=0)
+    dxy = dx + dy - origin
```
Comment (hnyu) [inline alf/utils/visualizer.py:85]: why dxy is computed in this way?

File: alf/utils/visualizer.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    # create the mini-image: [B, d, 2, 2], where d denotes the action dimensions
+    # [origin, dx]
+    # [dy,     dxy]
+    mini_image = torch.stack((torch.stack(
```
Comment (hnyu) [inline alf/utils/visualizer.py]: Why mini_image has a batch size of B here?

File: alf/utils/visualizer.py (hunk 1 above)
Comment (Haichao-Zhang) [reply alf/utils/visualizer.py]: changed

File: alf/utils/visualizer.py (hunk 2 above)
Comment (Haichao-Zhang) [reply alf/utils/visualizer.py]: Now changed the names to be more informative.

File: alf/utils/visualizer.py (hunk 4 above)
Comment (Haichao-Zhang) [reply alf/utils/visualizer.py]: There should be no ``B`` here. Removed.

**Review by hnyu — APPROVED** [review]

**Pull request merged**